src/Entity/CivicrmEntity.php

```diff
@@ -17,13 +22,98 @@
  */
 class CivicrmEntity extends ContentEntityBase {
 
+  /**
+   * {@inheritdoc}
+   */
   public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     $fields = [];
+    $civicrm_entity_info = SupportedEntities::getInfo()[$entity_type->id()];
+    $civicrm_required_fields = $entity_type->get('civicrm_required_fields');
     $field_definition_provider = \Drupal::service('civicrm_entity.field_definition_provider');
-    $civicrm_fields = \Drupal::service('civicrm_entity.api')->getFields($entity_type->get('civicrm_entity'));
+    $civicrm_fields = \Drupal::service('civicrm_entity.api')->getFields($entity_type->get('civicrm_entity'), 'create');
     foreach ($civicrm_fields as $civicrm_field) {
+      // If the entity has any field overrides, merge them in.
+      if (!empty($civicrm_entity_info['fields'][$civicrm_field['name']])) {
+        $civicrm_field = $civicrm_entity_info['fields'][$civicrm_field['name']] + $civicrm_field;
```

Comment:
This actually doesn't merge, but just fills missing values. Comment is misleading.